cylc/flow/workflow_db_mgr.py

```diff
@@ -506,7 +507,9 @@ def put_task_pool(self, pool: 'TaskPool') -> None:
                     "time_updated": itask.state.time_updated,
                     "submit_num": itask.submit_num,
                     "try_num": itask.get_try_num(),
-                    "status": itask.state.status
+                    "status": itask.state.status,
+                    "is_manual_submit": itask.is_manual_submit,
+                    "is_complete": itask.is_complete()
```

Comment:
Do we need an `is_complete` field?

Is this to protect us against the completion condition being changed later by reload/restart?

Reply:
No (although that sounds like something we need to consider).

We can now set outputs manually in "abstract" tasks, which means completed tasks can exist on disk even if they were never spawned before. So when asked to spawn a task, it's no longer sufficient to check that it was never spawned before in this flow. We have to check that it is not already complete in this flow. 

In principle we could work out every time we want to spawn a task: read recorded outputs from the DB and work out completion status, but it is a hell of a lot easier to just record completion status in the DB.